tests/examples/source/manifests/deploy_deployment_nodeport.yaml

```diff
@@ -0,0 +1,60 @@
+---
+    kind: Deployment
+    apiVersion: apps/v1
+    metadata:
+      name: COMPONENT_NAME
```

Comment:
I do see [this](https://github.com/openshift/odo/pull/3368/files#diff-369212aa33c85093d82cc8b496910026R126) featuring in the merged proposal, I wonder if we could use something more standardized with respect to template variables? 

Could we use `${COMPONENT_NAME}` instead of `COMPONENT_NAME` ? 
* That would adhere to how OpenShift Template yamls are written
* That would adhere to how Helm Templates are written
* You could use a Go template parser out-of-the-box

Reply:
Or, we could do a {{ .COMPONENT_NAME }} instead of COMPONENT_NAME and use a [Go Template Parser](https://golang.org/pkg/text/template/)

Trying to see if we can use a  more common way of templating this .

Reply:
@sbose78 Yes, we were contemplating changing this but hadn't come up with a suitable proposal.
If you think  that `{{ .COMPONENT_NAME }}` is a suitable way to go then we can look at modifying the code to support this format.

Reply:
`{{ .COMPONENT_NAME }}` seems to be the most standard way of doing things in the Kubernetes world, what do you think @kadel  ?

Reply:
+1 use Go Template Parser.